pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -176,22 +140,51 @@ func newGCECloud(config io.Reader) (*GCECloud, error) {
 			tokenSource = newAltTokenSource(cfg.Global.TokenURL, cfg.Global.TokenBody)
 		}
 	}
+
+	return CreateGCECloud(projectID, zone, networkURL, tokenSource)
+}
+
+// Creates a GCECloud object using the specified parameters.
+// If no networkUrl is specified, loads networkName via rest call.
+// If no tokenSource is specified, uses oauth2.DefaultTokenSource.
+func CreateGCECloud(projectID, zone, networkURL string, tokenSource oauth2.TokenSource) (*GCECloud, error) {
+	if tokenSource == nil {
+		var err error
+		tokenSource, err = google.DefaultTokenSource(
+			oauth2.NoContext,
+			compute.CloudPlatformScope,
+			compute.ComputeScope,
+			compute.DevstorageFullControlScope,
+			compute.DevstorageReadWriteScope)
```

Comment:
The devstorage.full_control scope is a superset of the read-write scope, so read-write isn't needed if full control is set. I'm frankly not sure why devstorage is needed at all, though, since the devstorage scopes only apply to GCS, which I don't believe we do anything with in this file.